tldw_Server_API/app/core/AuthNZ/repos/prototype_workspaces_repo.py

```
@@ -0,0 +1,805 @@
+@dataclass
+class PrototypeWorkspacesRepo:
+    """Data access for prototype workspace metadata in the AuthNZ DB."""
+
+    db_pool: DatabasePool
+
+    def _ts(self) -> datetime | str:
+        now = datetime.now(timezone.utc)
+        return now if getattr(self.db_pool, "pool", None) is not None else now.isoformat()
+
+    @staticmethod
+    def _row_to_dict(row: Any) -> dict[str, Any]:
```

Ruff (0.15.11)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `row`

(ANN401)